frontend/src/components/Navbar/NavBar.tsx

```diff
@@ -47,55 +60,56 @@ const Navbar: React.FC<NavbarProps> = ({ setIsCollapsed }) => {
                         </button>
                     </li>
 
                     {/* Hide Chat History Button */}
                     <li className="nav-item">
-                        <button className="btn btn-light btn-sm d-flex align-items-center gap-1">
+                        <button onClick={handleShowHistoryPanel} className="btn btn-light btn-sm d-flex align-items-center gap-1">
                             <IconHistory className={styles.iconLarge} />
-                            <span className="d-none d-md-inline">Hide chat history</span>
+                            <span className="d-none d-md-inline">{historyContent}</span>
                         </button>
                     </li>
 
                     {/* Settings Button */}
                     <li className="nav-item">
-                        <button className="btn btn-light btn-sm d-flex align-items-center gap-1">
+                        <button onClick={handleShowSettings} className="btn btn-light btn-sm d-flex align-items-center gap-1">
                             <IconSettings className={styles.iconLarge} />
                             <span className="d-none d-md-inline">Settings</span>
                         </button>
                     </li>
 
                     {/* User Profile Card */}
                     <li className="nav-item dropdown">
-                        <a className="nav-link" href="#" id="drop2" data-bs-toggle="dropdown" aria-expanded="false">
+                        <a className={`nav-link ${isDropdownOpen ? 'show' : ''}`} role="button" id="drop2" data-bs-toggle="dropdown" aria-expanded={isDropdownOpen} onClick={handleOnClickProfileCard}>
                             <div className={`d-flex align-items-center gap-2 ${styles.profileCard}`}>
                                 <IconBell className={`fs-6 ${styles.iconLarge}`} />
                                 <div className={styles.userDetails}>
-                                    <p className={`${styles.userName} mb-0`}>Manuel Castro</p>
-                                    <p className={`${styles.userEmail} mb-0`}>manuelcastro@hamalsolutions.com</p>
+                                    <p className={`${styles.userName} mb-0`}>{userName}</p>
+                                    <p className={`${styles.userEmail} mb-0`}>{email}</p>
                                 </div>
                             </div>
                         </a>
-                        <div className="dropdown-menu dropdown-menu-end dropdown-menu-animate-up" aria-labelledby="drop2">
+                        <div className={`dropdown-menu dropdown-menu-end animate-dropdown ${isDropdownOpen ? 'show' : ''}`} aria-labelledby="drop2" 
+                        data-bs-popper={`${isDropdownOpen ? 'static' : ''}`}>
                             <div className={styles.messageBody}>
                                 <a href="#" className="d-flex align-items-center gap-2 dropdown-item">
```

Comment:
@egdagger, could you replace the <a> element with the Link component from React Router to enable client-side navigation and better integrate with our routing system?